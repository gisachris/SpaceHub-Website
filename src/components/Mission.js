import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { joinMission, leaveMission } from '../redux/features/missionsSlice';

function Mission(props) {
  const dispatch = useDispatch();
  const { id, missionName, missionDesc } = props;
  const missions = useSelector((state) => state.missions.missions);
  const mission = missions.find((mission) => mission.mission_id === id);
  const joinClass = mission?.joined ? 'join-btn joined' : 'join-btn';
  const activateMember = mission.joined ? 'active-mission' : 'member';

  return (
    <tbody>
      <tr>
        <td className="name">{missionName}</td>
        <td>{missionDesc}</td>
        <td className="members">
          <span className={activateMember}>
            {mission.joined ? 'Active member' : 'Not a member'}
          </span>
        </td>
        <td>
          <button
            className={joinClass}
            onClick={() => (mission.joined
              ? dispatch(leaveMission(id))
              : dispatch(joinMission(id)))}
            type="submit"
          >
            {mission?.joined ? 'Leave mission' : 'Join mission'}
          </button>
        </td>
      </tr>
    </tbody>
  );
}

Mission.propTypes = {
  id: PropTypes.string.isRequired,
  missionName: PropTypes.string.isRequired,
  missionDesc: PropTypes.string.isRequired,
};

export default Mission;
