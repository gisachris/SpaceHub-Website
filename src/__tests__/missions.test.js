/*eslint-disable*/
import {render} from '@testing-library/react';
import Mission from '../mocks/Mission.mock';

describe('making tests for rocket components in profile view', () => {
  test('checking snapshot for rockets View in profile', () => {
    const missionData = [
      {
        id: 1,
        name: 'rocket 1',
        image: 'image1',
        description: 'description of rocket1',
        join: false,
      },
      {
        id: 2,
        name: 'rocket 2',
        image: 'image2',
        description: 'description of rocket2',
        join: false,
      },
      {
        id: 3,
        name: 'rocket 3',
        image: 'image3',
        description: 'description of rocket3',
        join: true,
      },
      {
        id: 4,
        name: 'rocket 4',
        image: 'image4',
        description: 'description of rocket4',
        join: true,
      },
    ];

    const joinData = missionData.filter(obj => obj.join === true);

    const {container} = render(<Mission joinData={joinData} />);
    expect(container).toMatchSnapshot();
  });
});
