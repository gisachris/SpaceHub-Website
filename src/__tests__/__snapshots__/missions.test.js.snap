// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`making tests for rocket components in profile view checking snapshot for rockets View in profile 1`] = `
<div>
  <section>
    <div
      class="mission"
    >
      <h2
        class="missionHeader"
      >
        mission
      </h2>
      <table
        class="holderTable"
      >
        <tbody>
          <tr
            class="missionrow"
          >
            <td
              class="missionData"
            >
              rocket 3
            </td>
          </tr>
          <tr
            class="missionrow"
          >
            <td
              class="missionData"
            >
              rocket 4
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</div>
`;
